src/reaction.jl

```diff
@@ -134,19 +138,19 @@ Notes:
 - The three-argument form assumes all reactant and product stoichiometric coefficients
   are one.
 """
-struct Reaction{S, T}
+struct Reaction{T}
     """The rate function (excluding mass action terms)."""
     rate::Any
     """Reaction substrates."""
-    substrates::Vector
+    substrates::Vector{BasicSymbolic{Real}}
```

Comment:
We should know that species are `BasicSymbolic{Real}`.

Technically, someone could declare a constantspecies parameter, and set it to a non-default type (i.e. `BasicSymbolic{Int64}`). However, we probably should not permit this?

Reply:
Let me know when you've reverted this and updated the PR as we discussed.

Reply:
The decision is to change to
```julia
substrates::Vector{S}
products::Vector{R}
```
for now

(sorry, was a bit unsure what the final conclusion was if there was one, more than happy to make this switch)

Reply:
No, it was to just leave them untyped as they currently are.